config/install/field.field.media.image.field_image_storage.yml

```diff
@@ -0,0 +1,27 @@
+uuid: 1a7e8c9e-d18f-42a6-8e7c-632c9dde5aaf
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.media.field_image_storage
+    - media_entity.bundle.image
+  module:
+    - file
+id: media.image.field_image_storage
+field_name: field_image_storage
+entity_type: media
+bundle: image
+label: 'Image Storage'
+description: ''
+required: false
+translatable: false
+default_value: {  }
+default_value_callback: ''
+settings:
+  file_directory: '[date:custom:Y]-[date:custom:m]'
+  file_extensions: 'png gif jpg jpeg tiff tif jp2'
```

Comment:
I did some tests, and for some reason, the field isn't obeying these values.  I don't think it's anything you've done.  It's probably Drupal.  Let's leave the defaults (`png gif jpg jpeg`) until I can take a deeper dive and figure out what's going on.